### PR TITLE
[NET-653] Improved Search for Public ID

### PR DESCRIPTION
**Pull request by doblanken**
[NET-653](https://nethinks.atlassian.net/browse/NET-653)

### CODE REVIEW

File: app/src/app/search/search-bar/search-bar.component.ts [hunk 1]
```diff
@@ -168,9 +161,9 @@ export class SearchBarComponent implements OnInit, OnDestroy {
         tag.searchLabel = params.length === 1 ? params[0].label : searchTerm;
         tag.settings = { categories: categoryIDs } as SearchBarTagSettings;
         break;
-      case 'publicID':
-        tag.settings = { publicID: searchTerm } as SearchBarTagSettings;
-        break;
+      // case 'publicID':
```
Comment (markheumueller) [inline app/src/app/search/search-bar/search-bar.component.ts:164]: You can remove the hidden comments here.

File: app/src/app/search/search.component.ts [hunk 2]
```diff
@@ -81,6 +81,7 @@ export class SearchComponent implements OnInit, OnDestroy {
    * List of search results.
    */
   public searchResultList: SearchResultList;
+  public publicIdResult = undefined;
```
Comment (markheumueller) [inline app/src/app/search/search.component.ts:84]: You don't have to initialize something with undefined unless you are overriding a value.

File: app/src/app/search/search.component.ts [hunk 3]
```diff
@@ -133,23 +134,39 @@ export class SearchComponent implements OnInit, OnDestroy {
    * Triggers the actual search api call.
    */
   public onSearch(): void {
+    this.publicIdResult = undefined;
```
Comment (markheumueller) [inline app/src/app/search/search.component.ts:137]: In this case it´s ok

File: app/src/app/search/search.component.ts [hunk 4]
```diff
@@ -133,23 +134,39 @@ export class SearchComponent implements OnInit, OnDestroy {
...
+    this.searchService.postSearch(this.queryParameters, params).pipe(takeUntil(this.subscriber)).subscribe((results: SearchResultList) => {
+        this.searchResultList = results;
+        this.filterResultList = this.initFilter && results !== null ? results.groups : this.filterResultList;
+        if (this.initSearch) {
+          this.maxNumberOfSites = Array.from({length: (this.searchResultList.total_results)}, (v, k) => k + 1);
+          this.initSearch = false;
+          this.initFilter = false;
+        }
     });
+    let localParameters = JSON.parse(this.queryParameters);
```
Comment (markheumueller) [inline app/src/app/search/search.component.ts:154]: You could handle the spinner case with a
`.add(() => this.spinner.hide());`
At the end of the subscription. Little bit nicer, but optional.

**Review by markheumueller — CHANGES_REQUESTED** [review]
The rest looks good.

File: app/src/app/search/search-bar/search-bar.component.ts [hunk 5]
```diff
@@ -148,6 +138,9 @@ export class SearchBarComponent implements OnInit, OnDestroy {
     switch (searchForm) {
       case 'text':
         tag.searchText = searchTerm.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&');
+        if (!isNaN(Number(searchTerm))) {
```
Comment (johndgrey) [inline app/src/app/search/search-bar/search-bar.component.ts:141]: I would extend this check by checking for integers. At the moment, float values are also allowed.

For Example like: 
 if (!isNaN(searchTerm) && Number.isInteger(searchTerm)) {

**Review by markheumueller — APPROVED** [review]

**Pull request merged**